dd-sdk-android-rx/src/test/kotlin/com/datadog/android/rx/DatadogRxExtTest.kt

```diff
@@ -0,0 +1,131 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rx
+
+import com.datadog.android.fresco.utils.Configurator
+import com.datadog.android.rum.GlobalRum
+import com.datadog.android.rum.RumErrorSource
+import com.datadog.android.rum.RumMonitor
+import com.datadog.tools.unit.getStaticValue
+import com.nhaarman.mockitokotlin2.verify
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.annotation.Forgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import io.reactivex.rxjava3.internal.operators.completable.CompletableError
+import io.reactivex.rxjava3.internal.operators.flowable.FlowableError
+import io.reactivex.rxjava3.internal.operators.maybe.MaybeError
+import io.reactivex.rxjava3.internal.operators.observable.ObservableError
+import io.reactivex.rxjava3.internal.operators.single.SingleError
+import java.util.concurrent.atomic.AtomicBoolean
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(
+        MockitoExtension::class,
+        ForgeExtension::class
+    )
+)
+@ForgeConfiguration(value = Configurator::class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+class DatadogRxExtTest {
+
+    @Mock
+    lateinit var mockRumMonitor: RumMonitor
+
+    @Forgery
+    lateinit var fakeException: Throwable
+
+    @BeforeEach
+    fun `set up`() {
+        GlobalRum.registerIfAbsent(mockRumMonitor)
+    }
+
+    @AfterEach
+    fun `tear down`() {
+        val isRegistered: AtomicBoolean = GlobalRum::class.java.getStaticValue("isRegistered")
+        isRegistered.set(false)
+    }
+
+    @Test
+    fun `M send an error event W exception in the stream {Observable}`(forge: Forge) {
+        // WHEN
+        ObservableError.error<Int>(fakeException).publishErrorsToRum().test()
```

Comment:
Nice, I'd probably still try to have a stream with at least a couple of valid values before emitting an error. That way we can be sure that the ext don't swallow any value. WDYT ?

Reply:
Hmm...that might be a nice idea, I am going to try to see how I can tweak that Observable to do that.